unittests/Solver/BasicSolverTests.cpp

```diff
@@ -0,0 +1,141 @@
+#include "vara/Solver/Solver.h"
+
+#include "z3++.h"
+
+#include "gtest/gtest.h"
+
+namespace vara::solver {
+
+TEST(Solver, Z3BasicTest) {
+  z3::context C;
+
+  z3::expr A = C.bool_const("A");
+  z3::expr B = C.bool_const("B");
+  z3::expr Expression = A && B;
+
+  z3::solver S(C);
+
+  S.add(Expression);
+  EXPECT_EQ(S.check(), z3::check_result::sat);
+
+  z3::expr InvalidatingExpression = !A;
+  S.add(InvalidatingExpression);
+  EXPECT_EQ(S.check(), z3::check_result::unsat);
+}
+
+TEST(Solver, RetrieveSatisfyingModel) {
+  z3::context Con;
+
+  z3::expr A = Con.bool_const("A");
+  z3::expr B = Con.bool_const("B");
+  z3::expr C = Con.bool_const("C");
+  z3::expr D = Con.bool_const("D");
+
+  z3::expr Expression = A && B && !C;
+
+  z3::solver S(Con);
+
+  S.add(Expression);
+  EXPECT_EQ(S.check(), z3::check_result::sat);
+
+  z3::model M = S.get_model();
+
+  // Retrieve the model by using the declaration function of each expression
+  EXPECT_TRUE(M.get_const_interp(A.decl()));
+  EXPECT_TRUE(M.get_const_interp(B.decl()));
+  EXPECT_FALSE(M.get_const_interp(C.decl()));
+
+  // Add the current configuration as a constraint
+  if (M.get_const_interp(D.decl())) {
```

Comment:
If I remember correctly, it could be that the solver (depending on the seed) could find a configuration in different order. But I will try to control this and remove the if in this test.